### PR TITLE
SOF-1792 Add/Remove Pieces when Segments/Parts are being updated.

### PR DESCRIPTION
**Pull request by LindvedKrvang**
Pieces are now being correctly added/removed when we receive updates to Segments and/or Parts.

If we receive and Ingest update event for a Segment or Part that had newly added or removed Pieces, these Pieces were either not added or removed. They are now.

### CODE REVIEW

**Review by RasmusAlbrektsen — APPROVED** [review]

File: src/data-access/repositories/interfaces/piece-repository.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export interface PieceRepository {
   getPiecesFromIds(pieceIds: string[]): Promise<Piece[]>
   savePiece(piece: Piece): Promise<void>
   deletePiecesForPart(partId: string): Promise<void>
+  deletePieces(pieceIdsToBeDeleted: string[]): Promise<void>
```
Comment (vivanov90) [inline src/data-access/repositories/interfaces/piece-repository.ts:8]: When the function name is `deletePieces`, is it imperative that the parameter name be `pieceIdsToBeDeleted`. Isn't it clear from the function name that `pieceIds` will be deleted.

File: src/data-access/repositories/mongo/mongo-piece-repository.ts [hunk 2]
```diff
@@ -56,6 +56,10 @@ export class MongoPieceRepository extends BaseMongoRepository implements PieceRe
     }
   }
 
+  public async deletePieces(pieceIdsToBeDeleted: string[]): Promise<void> {
```
Comment (vivanov90) [inline src/data-access/repositories/mongo/mongo-piece-repository.ts:59]: Same as that above.

**Review by vivanov90 — APPROVED** [review]

File: src/data-access/repositories/interfaces/piece-repository.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/data-access/repositories/interfaces/piece-repository.ts]: I would agree. I keep going back and forth on it. It's been updated :)

File: src/data-access/repositories/mongo/mongo-piece-repository.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/data-access/repositories/mongo/mongo-piece-repository.ts]: Same answer as above :)

**Pull request merged**